elastic/kibana
# [Security Solution] Consider exceptions when loading threshold alert timelines

## Summary
Fixes https://github.com/elastic/kibana/issues/126851

To test:
1. Create a Threshold Rule and add an exception.
2. Index some documents that will trigger the rule. Include at least one event that matches the exception.
3. Wait for alerts to be generated.
4. View details of the alert and observe the Threshold Count (the count of events matching the threshold rule over the configured interval + lookback).
5. Now view the alert in Timeline. Observe that the number of events equals the Threshold Count from previous step. A filter should be populated with the exceptions (within an alias):

<img width="1318" alt="image" src="https://user-images.githubusercontent.com/611653/160693274-77ef896b-65a0-47d2-9a88-9a34c371e53a.png">



### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to identify risks that should be tested prior to the change/feature release.

When forming the risk matrix, consider some of the following examples and how they may potentially impact the change:

| Risk                      | Probability | Severity | Mitigation/Notes        |
|---------------------------|-------------|----------|-------------------------|
| Multiple Spaces&mdash;unexpected behavior in non-default Kibana Space. | Low | High | Integration tests will verify that all features are still supported in non-default Kibana Space and when user switches between spaces. |
| Multiple nodes&mdash;Elasticsearch polling might have race conditions when multiple Kibana nodes are polling for the same tasks. | High | Low | Tasks are idempotent, so executing them multiple times will not result in logical error, but will degrade performance. To test for this case we add plenty of unit tests around this logic and document manual testing procedure. |
| Code should gracefully handle cases when feature X or plugin Y are disabled. | Medium | High | Unit tests will verify that any feature flag or plugin combination still results in our service operational. |
| [See more potential risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx) |


### For maintainers
